### PR TITLE
Reducer and action

### PR DESCRIPTION
**Pull request by abenaTenkorang**
**In this pull request, the following changes are made**

-  Install redux.
-  Create reducers and actions for books
-  Create reducers and actions for categories.
-  Create a config function for store.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      return {
+        ...state,
+        books: [...state.books.filter((book) => book.id !== payload)],
+      };
+    default:
+      return state;
+  }
+};
+
+export default books;
```
Comment (rloterh) [inline src/redux/books/booksSlice.js]: - [ ] In this activity we want you to simplify the generation of _action_ _types_ and _creators_ by using the `createSlice` from `@reduxjs/toolkit` package. Also, ensure to:
  * Exports by default `booksSlice.reducer`
  * Exports by `addBook` and `removeBook` as actions. 
  
  Example is as below:
```javascript
// ./src/redux/books/booksSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const booksSlice = createSlice({
  name: "books",
  initialState: {
    books: [],
  },
  reducers: {
    addBook: (state, action) => {
      state.books = [
        ...state.books,
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
        },
      ]
    },
    removeBook: (state, action) => {
      state.books = state.books.filter((book) => book.id !== action.id)
    },
  },
})

export const { addBook, removeBook } = booksSlice.actions

export default booksSlice.reducer
```

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        ...state,
+        categories: payload,
+      };
+
+    default:
+      return state;
+  }
+};
+
+export default categories;
```
Comment (rloterh) [inline src/redux/categories/categoriesSlice.js]: - [ ] Similarly, the generation of _action_ _types_ and _creators_ should be done by using the `createSlice` from `@reduxjs/toolkit` package. Also, let's ensure to:
  * Exports by default `categoriesSlice.reducer`
  * Exports by `checkStatus` as actions. 
  
  Example is as below:
```javascript
// ./src/redux/categories/categoriesSlice.js

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Abi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 🏃‍♀️ 🌴 🚗 🏠 🌳 🏡 🏎️ 

## Highlights 👍  
- Good job so far implementing action state management using _store_, _actions_ and _reducers_!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷 🇬🇭 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ABDULALI3468 — APPROVED** [review]
Hi @abenaTenkorang , 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 
✅Well implemented redux store!
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**